web/src/containers/Logout.js

```diff
@@ -0,0 +1,22 @@
+import PropTypes from 'prop-types';
```

Comment:
An empty page that simply fires the logout action.  This lets us create a route for logging out, and we can intercept routing behavior to alert about unsaved changes.  This is much easier than trying to modify the logout behavior to be internally conditional.